### PR TITLE
Change location to three fields

### PR DESCRIPTION
**Pull request by threedaymonk**
* Location in building
* Building
* City

The location field has been renamed to location_in_building, so existing profiles will have all information in this field.

When viewing a profile, the fields are concatenated with commas (with blank fields omitted).

### CODE REVIEW

File: app/models/peoplefinder/concerns/completion.rb [hunk 1]
```diff
@@ -21,7 +15,7 @@ def completion_score_fields
       lambda {
         where(%{
           COALESCE(image,'') = ''
-          OR COALESCE(location,'') = ''
+          OR COALESCE(location_in_building,'') = ''
```
Comment (threedaymonk) [inline app/models/peoplefinder/concerns/completion.rb]: I don't know whether this is appropriate for determining "inadequate profiles": I'm not sure how this is used.

File: app/views/peoplefinder/people/_form.html.haml [hunk 2]
```diff
@@ -51,11 +51,22 @@
...
+        = f.label :building, class: 'form-label-bold' do
+          Building
+          %p.form-hint= info_text('building_hint')
+        = f.text_field :building, class: 'form-control'
+
+      .form-group
+        = f.label :city, class: 'form-label-bold' do
+          City
+          %p.form-hint= info_text('city_hint')
+        = f.text_field :city, class: 'form-control'
```
Comment (threedaymonk) [inline app/views/peoplefinder/people/_form.html.haml]: We discussed placeholder text, but I elected to use hints as we were already doing that. (And hints have the nice property of still being visible after you've started typing.)

**Comment by threedaymonk** [comment]
Editing:

screenshot-2015-02-13t12 00 09z

Viewing:

screenshot-2015-02-13t12 00 35z

File: app/models/peoplefinder/concerns/completion.rb (hunk 1 above)
Comment (novotnyjakub) [inline app/models/peoplefinder/concerns/completion.rb]: Didn't we say that all the location fields have to be filled for the profile to be complete? I'll ask Jon.

File: app/views/peoplefinder/people/_form.html.haml (hunk 2 above)
Comment (novotnyjakub) [inline app/views/peoplefinder/people/_form.html.haml]: :+1:

File: db/migrate/20150213103214_add_additional_location_fields.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddAdditionalLocationFields < ActiveRecord::Migration
+  def change
+    rename_column 'people', 'location', 'location_in_building'
+    add_column 'people', 'building', :text
+    add_column 'people', 'city', :text
```
Comment (novotnyjakub) [inline db/migrate/20150213103214_add_additional_location_fields.rb]: I just noticed that we're mixing strings and symbols in our migrations. But it's no biggie.

File: app/models/peoplefinder/concerns/completion.rb (hunk 1 above)
Comment (threedaymonk) [inline app/models/peoplefinder/concerns/completion.rb]: This is different from the percentage complete, though.

Comment (novotnyjakub) [inline app/models/peoplefinder/concerns/completion.rb]: I know, this is used for getting a list of people who to send a reminder e-mail to - for them to complete the profile. So it should be inline with the completeness and if somebody doesn't have a city filled in (which is counted against completion score), this scope should also retrieve such person.

**Comment by Najaf** [comment]
Also happy with this. @threedaymonk do you want to do some rebasing or something before I click merge?

**Pull request merged**